### DISCUSSION TITLE
How to add/enable Connection Object (ID 0x05)?

### DISCUSSION THREAD

**Discussion by troglobit** [EIPStackGroup/OpENer#388] [Q&A] [answered]
I'm still very new at CIP so please forgive me for this possibly stupid question.  How do I add or enable the Connection Object (ID 0x05) to the adapter?

There is mention of it in the code https://eipstackgroup.github.io/OpENer/d1/d6c/cipconnectionobject_8h.html for instance `#define CIP_CONNECTION_OBJECT_CODE 0x05`
but it doesn't seem to be enabled by default.

**Comment by MartinMelikMerkumians** [comment, marked as answer]
Hi,

I had the same question when I started with EtherNet/IP, that's a perfect and reasonable question.
The Connection Object is not accessible from outside of OpENer, as it supports the (Large) Forward Open services, which handles Connection Object creation and deletion.
Therefore there is not really a need to expose connection object in such a case, and as far as I know most stacks also dont expose connection objects.
I also think there are some restrictions what the connection object is allowed to do if the device supports Forward Open.

Not sure if this fully answers your question.

**Reply by troglobit** [reply]
That's a great answer, and just what I suspected. Thank you so much! I've discussed the request with my customer and they agree it's not needed (not a mandatory CIP object to support anyway).
